### DISCUSSION TITLE
Guide for adding user login, social login or SSO

### DISCUSSION THREAD

**Discussion by yasirmturk** [lunarphp/lunar#2018] [Q&A]
Is there any recommended way to add user auth and social login features to lunar which is most compatible and easy to integrate? Laravel itself provides multiple ways to do this; I want to know a general recommendation and one specific to the livewire-starter-kit

Thanks in advance

**Comment by glennjacobs** [comment]
Not really, it's just Laravel. Just remember that Lunar also requires a related Customer on users.

**Reply by yasirmturk** [reply]
yes, that is what i meant. I want the customers to be able to login and see order history and other stuff

**Comment by xalunda** [comment]
Hi there!

I wanted to do a "customer registration" endpoint too, did you figure out an easy way to do this?

As Lunar implemented a `staff` system for admin users, I'm guessing that we can use the `users` directly without repercussion.
Could a maintainer confirm this?

Thanks!

**Reply by glennjacobs** [reply]
You wouldn't want to use the Staff model, as that's for administrators.

You should just use the standard Laravel User model as you would in most Laravel applications. The only difference is, you'll also want to create a `Lunar\Models\Customer` entry also that is associated to the user you create.

**Reply by xalunda** [reply]
> You wouldn't want to use the Staff model, as that's for administrators.

Yes that part was clear, thanks for confirming.

What I dont understand is the reason why the `User` is not created by Lunar along with the `Customer`. 
What's the reason for not doing it automatically if I may ask?

**Reply by glennjacobs** [reply]
Primarily flexibility. Projects require different requirements. Some might have multiple users associated to a customer for example. As it's basic Laravel we leave that well alone and allow the developer to decide. Lunar tries to be as minimally opinionated as possible.

**Reply by yasirmturk** [reply]
@glennjacobs flexibility is understandable but it is kind of disabling. 
store is forced 
- to create a customer with duplicate information when user is created
- to implement logic to verify that if there is a user then it should have a customer
- to prevent having dangling `users` in the system which is very much possible
- a lot of if and null checks because user can have many customers and customers can have many users 
- `$user->latestCustomer()` is bit confusing, it could be `$user->selectedCustomer()` or `$user->currentCustomer()` or `$user->defaultCustomer()` and not null 
- to ask the user to select a customer 

Now, i am not suggesting to remove or change this design 
but 
there should be an implementation which should be able to handle the default single users single customer scenario. for example apis for 
- creating a customer (+addresses) from an order, or user
- get a product discount for specific user/customer
- hook/events to handle dangling cases etc.

**Comment by yasirmturk** [comment]
@xalunda how I managed this is to install laravel breeze in a new example project and then copy all the auth related changes to my project. 
- This is will bring in all the user and auth functions
- Then I modified the register form to change from laravel full name to lunar first name last name fields 
- Then I modifier the register function to created a customer after the user has been created
- Override implementation of `AssociateUser` to cart 
- and some more tweaks here and there you can see in my fork https://github.com/ApniSite/aurastyle
